legwork/source.py

```diff
@@ -43,6 +45,26 @@ class Source():
         Semi-major axis (either `a` or `f_orb` must be supplied). Must have
         astropy units of length.
 
+    phi : `float/array`
+        azimuth of the source in the spherical projection on the sky. Must
+        have astropy angular units
+
+    theta : `float/array`
+        inclination from the pole of the source in the spherical
+        projection on the sky. Must have astropy angular units
```

Comment:
I think the comments for phi/theta and ra/dec should mention that one of the combinations is required and which one supersedes the other

Reply:
This one isn't necessary (and happy to do it since it would be finicky) but I think this could be more readable if we instead remove `phi, theta, ra, dec` and replace it with something like `position, coordinates` where `position=(x, y)` is a tuple of either ra, dec or theta, phi and `coordinates` is a string that says either "radec" or "thetaphi"

I think it is better because the variable names are more immediately descriptive. Just personally reading through this I get all of the psi, phi, thetas confused.